### PR TITLE
Add txpool_content rpc method for pending txs fetch

### PR DESCRIPTION
**Pull request by g11tech**
Extracted from shandong
- https://github.com/ethereumjs/ethereumjs-monorepo/pull/2316

developed by @acolytec3 

This allows blockscout to inspect txpool and fetch the pending transactions to list in explorer

TODO:
- [x] add the spec test

### CODE REVIEW

File: packages/client/lib/rpc/modules/txpool.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  /**
+   * Returns the contents of the transaction pool
+   * @param params An empty array
+   */
+  content(_params = []) {
+    const pending = new Map()
+    for (const pool of this._txpool.pool) {
+      const pendingForAcct = new Map<bigint, any>()
+      for (const tx of pool[1]) {
+        pendingForAcct.set(tx.tx.nonce, tx.tx.toJSON())
```
Comment (jochem-brouwer) [inline packages/client/lib/rpc/modules/txpool.ts:35]: Can't we just `pool[1].map((e) => e.tx.toJSON()`?

**Comment by acolytec3** [comment]
I've fixed a missing export so the new namespace is now exposed and have added a test.  Should be pretty close to ready at this point.

File: packages/client/lib/rpc/modules/txpool.ts (hunk 1 above)
Comment (acolytec3) [reply packages/client/lib/rpc/modules/txpool.ts:35]: Yes, yes we can.  :-)  Good catch.

**Comment by acolytec3** [comment]
Looks like it breaks something in the client tests though.  I can take a further look tomorrow as needed.

**Comment by acolytec3** [comment]
The fix in almost all cases on the failing client tests is adding an `includeVM:true` to the client constructor in the rpc tests.  I can take a stab at this tomorrow.

**Comment by acolytec3** [comment]
@holgerd77 Any opinions on this commit?  602f8b6903ae0c70b722304ca8cbc69a642fff1d  I don't know the history here as to why we don't always instantiate the VM when we're doing this client setup for the rpc tests.  The idea of this commit is to include the VM by default since otherwise I have to make tons of references to `createClient({ includeVM: true})`.  We need this because the `txpool_content` RPC class requires a VM in the constructor.  We can probably take different approaches if if desired bso don't want to merge this without further discussion.

**Comment by holgerd77** [comment]
> @holgerd77 Any opinions on this commit? [602f8b6](https://github.com/ethereumjs/ethereumjs-monorepo/commit/602f8b6903ae0c70b722304ca8cbc69a642fff1d) I don't know the history here as to why we don't always instantiate the VM when we're doing this client setup for the rpc tests. The idea of this commit is to include the VM by default since otherwise I have to make tons of references to `createClient({ includeVM: true})`. We need this because the `txpool_content` RPC class requires a VM in the constructor. We can probably take different approaches if if desired bso don't want to merge this without further discussion.

Hmm, I just looked a bit in the RPC test `helper.ts` file where this gets applied: I have to say I also can't recall the complete history and reasoning for this. I would say since this is such an undocumented switch we can't easily find out anymore anyhow. So we should just do the switch (as you suggested) and see what happens. Might have some side effects on other tests (maybe we wanted to test in some other parts without VM?), but I have absolutely no idea or certainty here.

So, try out, rather make sure to *then* run the tests early on completely. Then we'll see.

**Comment by acolytec3** [comment]
> So, try out, rather make sure to _then_ run the tests early on completely. Then we'll see.

Done.  And, all the tests pass.  This feels merge ready unless @g11tech or @holgerd77 or someone else has another opinion.

> Hmm, I just looked a bit in the RPC test helper.ts file where this gets applied: I have to say I also can't recall the complete history and reasoning for this. I would say since this is such an undocumented switch we can't easily find out anymore anyhow. So we should just do the switch (as you suggested) and see what happens. Might have some side effects on other tests (maybe we wanted to test in some other parts without VM?), but I have absolutely no idea or certainty here.

I may follow this up tomorrow or the next day with a PR to clarify all of those `createClient` opts as an options dictionary so we know what we're looking at.

**Comment by acolytec3** [comment]
I tested this locally against blockscout and discovered we'd missed a couple of things from the Shandong branch so pending txns weren't showing up.  I've brought those over from Shandong and tested with blockscout and things are now working correctly.  This should be ready for final review.

**Review by g11tech — COMMENTED** [review]
looks pretty neat! @acolytec3 :heart:  you will have to approve/merge!

**Review by acolytec3 — APPROVED** [review]
Approving my own work but yay for this!

**Pull request merged**